subgraphs/mux-protocol/src/entities/protocol.ts

```diff
@@ -0,0 +1,384 @@
+    case EventType.Liquidated:
+      protocol.cumulativeClosedInflowVolumeUSD =
+        protocol.cumulativeClosedInflowVolumeUSD.plus(collateralUSDDelta);
+      break;
+    default:
+      break;
+  }
+
+  protocol._lastUpdateTimestamp = event.block.timestamp;
+  protocol.save();
+}
+
+export function increaseProtocolPremium(
+  event: ethereum.Event,
+  amountUSD: BigDecimal,
+  eventType: EventType
+): void {
+  const protocol = getOrCreateProtocol();
+  switch (eventType) {
+    case EventType.Deposit:
+      protocol.cumulativeDepositPremiumUSD =
+        protocol.cumulativeDepositPremiumUSD.plus(amountUSD);
+      protocol.cumulativeTotalLiquidityPremiumUSD =
+        protocol.cumulativeTotalLiquidityPremiumUSD.plus(amountUSD);
+      break;
+    case EventType.Withdraw:
+      protocol.cumulativeWithdrawPremiumUSD =
+        protocol.cumulativeWithdrawPremiumUSD.plus(amountUSD);
+      protocol.cumulativeTotalLiquidityPremiumUSD =
+        protocol.cumulativeTotalLiquidityPremiumUSD.plus(amountUSD);
+      break;
+    case EventType.CollateralIn:
+      protocol.cumulativeEntryPremiumUSD =
+        protocol.cumulativeEntryPremiumUSD.plus(amountUSD);
+      protocol.cumulativeTotalPremiumUSD =
+        protocol.cumulativeTotalPremiumUSD.plus(amountUSD);
+      break;
+    case EventType.CollateralOut:
+      protocol.cumulativeExitPremiumUSD =
+        protocol.cumulativeExitPremiumUSD.plus(amountUSD);
+      protocol.cumulativeTotalPremiumUSD =
+        protocol.cumulativeTotalPremiumUSD.plus(amountUSD);
+      break;
+
+    default:
+      break;
+  }
+
+  protocol._lastUpdateTimestamp = event.block.timestamp;
+  protocol.save();
+}
+
+export function updateProtocolTVL(
+  event: ethereum.Event,
+  tvlChangeUSD: BigDecimal
+): void {
+  const protocol = getOrCreateProtocol();
+  protocol.totalValueLockedUSD =
+    protocol.totalValueLockedUSD.plus(tvlChangeUSD);
+  protocol._lastUpdateTimestamp = event.block.timestamp;
+  protocol.save();
+}
+
+export function updateProtocolOpenInterestUSD(
+  event: ethereum.Event,
+  openInterestChangeUSD: BigDecimal,
+  isLong: boolean
+): void {
+  const protocol = getOrCreateProtocol();
+  if (isLong) {
+    protocol.longOpenInterestUSD =
+      protocol.longOpenInterestUSD.plus(openInterestChangeUSD) >=
+      BIGDECIMAL_ZERO
+        ? protocol.longOpenInterestUSD.plus(openInterestChangeUSD)
+        : BIGDECIMAL_ZERO;
+  } else {
+    protocol.shortOpenInterestUSD =
+      protocol.shortOpenInterestUSD.plus(openInterestChangeUSD) >=
+      BIGDECIMAL_ZERO
+        ? protocol.shortOpenInterestUSD.plus(openInterestChangeUSD)
+        : BIGDECIMAL_ZERO;
+  }
+  protocol.totalOpenInterestUSD = protocol.longOpenInterestUSD.plus(
+    protocol.shortOpenInterestUSD
+  );
+
+  protocol._lastUpdateTimestamp = event.block.timestamp;
+  protocol.save();
+}
+
+export function incrementProtocolEventCount(
+  event: ethereum.Event,
+  eventType: EventType,
+  sizeDelta: BigInt
```

Comment:
`sizeDelta` not in use here too

Reply:
It is used at https://github.com/messari/subgraphs/blob/b68898ed16191780328266281c242cfe2f7e8ad7/subgraphs/mux-protocol/src/entities/protocol.ts#L222